modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -0,0 +1,210 @@
+        User user = (User) objectUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Users");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getUser(id, requiredAttributes)).thenReturn(user);
+
+        SCIMResponse outputScimResponse = userResourceManager.get(id, userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(outputScimResponse.getResponseMessage());
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+        Assert.assertEquals(obj.getString("id"), id);
+    }
+
+    @DataProvider(name = "dataForGetThrowingExceptions")
+    public Object[][] dataToGetThrowingExceptions() {
+
+        UserManager userManager = mock(UserManager.class);
+
+        return new Object[][]{
+
+                {"1234", userManager, "userName", null}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetThrowingExceptions")
+    public void testGetThrowingExceptions(String id, Object objectUserManager, String attributes,
+                     String excludeAttributes) throws CharonException, BadRequestException, NotFoundException {
+
+        UserManager userManager = (UserManager) objectUserManager;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema)
+                        CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Users");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getUser(id, requiredAttributes)).thenReturn(null);
+
+        SCIMResponse outputScimResponse = userResourceManager.get(id, userManager, attributes, excludeAttributes);
+
+        Assert.assertNull(outputScimResponse);
+    }
+
+    @DataProvider(name = "dataForListWithGetInt")
+    public Object[][] dataToGetListInt() throws CharonException, BadRequestException, InternalErrorException {
+
+        getNewUser();
```

Comment:
Why we need to call this method in here?

Reply:
fixed in f6c70ca0d7276c5e56af08faad3962e0bb9df4ad